### ISSUE TITLE
Angular 9: Cannot create property 'next' on number '<number>'

### ISSUE THREAD

**Issue by PapaNappa** [angular/angular#35299]
# 🐞 bug report

### Affected Package
I cannot certainly pin-point a specific package, but 

 The issue is probably caused by package @angular/core

### Is this a regression?

 I do not know, I have not used Ivy in Angular 8.

### Description

When loading a certain component with i18n for the first time, Angular produces the error

```
ERROR TypeError: Cannot create property 'next' on number '1'
    at createTNodeAtIndex (core.js:11541)
    at getOrCreateTNode (core.js:11499)
    at createDynamicNodeAtIndex (core.js:34653)
    at readCreateOpCodes (core.js:34755)
    at i18nEndFirstPass (core.js:34616)
    at ɵɵi18nEnd (core.js:34594)
    at Module.ɵɵi18n (core.js:34961)
    at BComponent_mat_card_0_Template (b.component.html:14)
    at executeTemplate (core.js:11841)
    at renderView (core.js:11627)
    at TemplateRef.createEmbeddedView (core.js:15157)
    at ViewContainerRef.createEmbeddedView (core.js:15264)
    at common.js:4365
    at DefaultIterableDiffer.forEachOperation (core.js:27734)
    at NgForOf._applyChanges (common.js:4353)
```

This error seems to be caused by the template
```
{bar(x), plural, =1 {{{bar(x)}} day} other {{{bar(x)}} days} }
```
But this error is dependent on a lot of circumstances to occur.

## 🔬 Minimal Reproduction

I tried my best to create a minimal reproduction.
However, the error goes away when I change seemingly arbitrary parts of the app - not neccessarily related to the source of the original error (e.g. changing a `<button mat-button>` to `<button>` made the error go away).

My reproduction is available at https://github.com/PapaNappa/Angular9Bug

The code is split into three directories:

- `a-b/` contains the main components after login. `AComponent` is a wrapper around `BComponent`. `BComponent` contains the i18n template.
- `authentication/` contains the `LoginComponent` and everything else related
- `top-bar` contains two layout-related components. `TopBarComponent` contains just a `<mat-toolbar>`, whereas `MainTopbarLayoutComponent` displays the `TopBarComponent` and its `<ng-content>`.

`AComponent` uses the `MainTopbarLayoutComponent` to display `BComponent`.

For this bug report, this structure is quite complex.
But I could not reduce any of it's complexity, or otherwise the bug could not be reproduced anymore.

As an interesting side-note: The <number> in the error message started at 8092 or something like that, and when I gradually removed code from my original app, the number decreased from time to time, but was always a power of two. In the current state of the reproduction, the number is now 1.

### Steps to reproduce

1. `git clone https://github.com/PapaNappa/Angular9Bug`
2. `cd Angular9Bug`
3. `npm install`
4. `ng serve`
5. Open the app at http://localhost:4200/ and open the Developer Console
6. Click on the "Login" button on the bottom
7. Observe the error in the Console
8. Hit `F5` to reload the page - the error goes away
9. Do reproduce the error again, click "Logout" on the top, *then reload the page*, then continue with step 6.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.1
Node: 12.14.1
OS: win32 x64

Angular: 9.0.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, localize, material, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.1
@angular-devkit/build-angular     0.900.1
@angular-devkit/build-optimizer   0.900.1
@angular-devkit/build-webpack     0.900.1
@angular-devkit/core              9.0.1
@angular-devkit/schematics        9.0.1
@angular/cli                      9.0.1
@angular/flex-layout              9.0.0-beta.29
@ngtools/webpack                  9.0.1
@schematics/angular               9.0.1
@schematics/update                0.900.1
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2

</code></pre>

**Comment by tieppt** [comment]
You are using the format of i18n without the `i18n` directive, https://angular.io/guide/i18n#pluralization but the compiler still generates i18n code (should create text + interpolation IMHO).
If you place the directive into the `span` tag, the application will work. But the compiler should fix the code.

**Comment by AndrewKushnir** [comment]
Hi @PapaNappa, thanks for reporting the issue.

As @tieppt pointed out, the problem was triggered by the ICU ([in this template](https://github.com/PapaNappa/Angular9Bug/blob/master/src/app/a-b/b.component.html#L14)) that was placed outside of the i18n block. I've created a PR (https://github.com/angular/angular/pull/35347) to fix the issue with such ICUs.

Aalso as a possible solution you can create an i18n block explicitly, for example:

```
<ng-container i18n>
  {bar(x), plural, =1 {{{bar(x)}} day} other {{{bar(x)}} days} }
</ng-container>
```

This would work with the current 9.0.0 version (without the changes in the PR that I mentioned).

Thank you.

**Comment by AndrewKushnir** [comment]
Hi @PapaNappa, PR #35347 was merged into master and the changes are available in 9.0.2 version of the framework (which is available on NPM). Please upgrade to the 9.0.2 version to get the latest changes. Feel free to reopen this ticket if the problem still exists. Thank you.
